### ISSUE TITLE
Confusion regarding world saving/world management

### ISSUE THREAD

**Issue by shiftyscales** [Frooxius/NeosPublic#996]
There have been a few issues that have come up repeatedly with regards to world saving:
- A user is not in their local home, nor has their inventory open so both the 'save here/save to inventory' are not clickable
- A user has their inventory open to a folder they can't save to so the saving option is not clickable
- Users delete a world orb from their inventory, and assume they've deleted the world/are confused when their storage doesn't drop

I propose that when saving a world, it should save directly into the world browser instead of the inventory without the need for a prompt.

The world browser should be used as the primary source for managing owned/created worlds and copies thereof, including deleting worlds, adjusting their privacy settings/permissions/tags/other metadata.

The 'world orb' concept causes confusion because sometimes it just serves as a link to the world, whereas other times it is responsible for managing the world's properties, e.g. setting the metadata/access level, deleting the world, etc.

As a short-term fix for the first point listed- if during the save process, the inventory was auto-focused- either to the last access folder, or the root folder if saving is still inaccessible, it could help resolve a lot of the confusion regarding being unable to save the world.

Long-term it would be best to have world management options available through the world browser.

**Comment by Frooxius** [comment]
This is planned to be addressed when the Inventory itself is reworked, as the system will instead just prompt the user where do they want to save the world.

I don't think using the World Browser is a good solution, as it will make for a nightmare for organization, putting everything into a flat hierarchy. It will also add another mechanism to save things, because when saving items, you'll still be using inventory and I want those mechanisms to be consistent, reducing how much user has to learn.

**Comment by shiftyscales** [comment]
A lot of confusion stems from the fact that a world orb saved to the inventory isn't actually the world itself. It's just a link to the world.

The same isn't really true for items. Once you remove the item from your inventory, the storage space it takes up is cleared out if the assets it used are no longer associated to your account, and the item is effectively deleted unless a backup is saved elsewhere or in someone else's inventory.

With regards to organization- wouldn't the search take care of a majority of the needs with regards to finding a specific world?

Plus, having it default to automatically saving just to the world browser wouldn't exclude users from also saving a copy of the world orb to their inventory if they so choose to do so.

**Comment by Frooxius** [comment]
Yes, that'll be addressed by the inventory redesign, so you can save worlds directly, rather than just orb to them. The inventory would be also more aware of this - e.g. if you delete a reference to a world, it would ask you if you want to delete the world itself as well.

Search should work in conjunction with organization. If you're searching for particular world and know the term it works, but it doesn't provide a way of browsing when you don't know what exactly you need to search for. Think of file organization in OS - imagine all your files were dumped into a flat list and only way to navigate was to search.

E.g. you can have different versions of the same world, backup copies and so on. Or some related items in the same subfolder that you need to work on that world. As the number of them grows, search wouldn't be manageable in a lot of cases.

**Comment by shiftyscales** [comment]
Yeah, I can understand that. Heavier integration into the inventory itself would satisfy the needs of this issue.

For now though, would you consider the possibility of having the inventory screen be focused while initiating the world saving process to help mitigate some of the confusion it causes, @Frooxius?

It is labelled 'Save To Inventory' it would make sense that the inventory should be open during this process so the user can choose where to save it rather than needing to manually browse to the inventory after having selected that option.
